client/post-editor/editor-seo-accordion/index.jsx

```diff
@@ -113,9 +113,4 @@ const mapStateToProps = state => {
 	};
 };
 
-export default connect(
-	mapStateToProps,
-	null,
-	null,
-	{ pure: false } // defaults to true, but this component has internal state
```

Comment:
What do you think about the comment "this component has internal state"? Does it matter if a component has internal state or not for its pureness?

Reply:
I think that's misunderstanding of `{ pure: false}` by the author of that comment. The `connect`-ed component can have local state without any restrictions. No special options on the `connect` call are needed.

When using `connect`, there are two components involved, and each of them can have state:

```jsx
<Connect(Component)>
  <Component />
</Connect(Component)>
```
The outer component (`Connect(Component)`) is subcribed to the Redux store, and the result of `mapStateToProps` is set as its local state on every change in Redux. The state is an implementation detail that you don't usually see. The properties of this state are passed down as props to the inner component (`Component`).

The inner component has the outer's component's state as `props`, and its own state exists independently.

The outer component has `shouldComponentUpdate` generated by `connect()`, which shallow-compares the outer component's state and props and avoids rerender of the inner component if there is no change. That can be disabled by `{ pure: false }`.

If the inner component's state changes, however, the update is done directly on the inner component and the outer component is not involved at all -- it doesn't matter if it has `shouldComponentUpdate` or not.